File: app/src/main/java/com/github/fo2rist/mclaren/mvp/MainScreenContract.java
package com.github.fo2rist.mclaren.mvp;

import android.support.annotation.NonNull;

public interface MainScreenContract {

    interface View extends BaseView {
        void openStories();
        void openCircuits();
        void openDrivers();
        void openNewsFeed();
        void openAboutScreen();
        void openTransmissionCenter();

        void navigateTo(String externalUrl);

        void showTransmissionButton();
    }

    interface Presenter extends BasePresenter<View> {
        /** Should be called when activity re-created with state restored. */
        void onRestart(@NonNull View view);
        void onStoriesClicked();
        void onCircuitsClicked();
        void onDriversClicked();
        void onNewsFeedClicked();
        void onCarClicked();
        void onOfficialSiteClicked();
        void onAboutClicked();
        void onTransmissionCenterClicked();
    }
}

File: app/src/main/java/com/github/fo2rist/mclaren/mvp/MainScreenContract.kt
package com.github.fo2rist.mclaren.mvp

import org.joda.time.DateTime

/**
 * Contract for Main Screen with navigation.
 */
interface MainScreenContract {

    interface View : BaseView {
        fun openStories()
        fun openTweets()
        fun openCircuits()
        fun openDrivers()

        fun openTransmissionCenter()

        fun navigateToCircuitScreen(eventNumber: Int)
        fun navigateToAboutScreen()
        fun navigateTo(externalUrl: String)

        fun showTransmissionButton()
        fun showUpcomingEventButton(grandPrixName: String, beginningTime: DateTime)
        fun hideFloatingButtons()
    }

    interface Presenter : BasePresenter<View> {
        fun onStart()

        fun onStoriesClicked()
        fun onTeamTwitterClicked()
        fun onSeasonCalendarClicked()
        fun onDriversClicked()
        fun onCarClicked()
        fun onOfficialSiteClicked()
        fun onAboutClicked()
        fun onTransmissionCenterClicked()
        fun onUpcomingEventClicked()
    }
}
